api/v2/api.go

```diff
@@ -238,6 +249,28 @@ func (api *API) getAlertsHandler(params alert_ops.GetAlertsParams) middleware.Re
 			continue
 		}
 
+		// TODO: Add a timestamp now()+X, after which the Mutes() will
+		// be refreshed?
+		// Will check if the alert is inhibited, and update its status.
+		api.muter.Mutes(a.Labels)
+		// Will check if the alert is silenced, and update its status.
+		sils, err := api.silences.Query(
```

Comment:
If I understand correctly, this logic determines whether a given alert is silenced or not. If so, that domain logic should rather live in `github.com/prometheus/alertmanager/silence`. What do you think?

Reply:
Yup. this is now mostly (but not exactly) replicated from https://github.com/prometheus/alertmanager/blob/master/notify/notify.go#L407-L425
Perhaps this could be unified in some way?

Reply:
The `silences` package exposes this `Query` api so as to be externally consumed. It seems to me that this is being used correctly. If we want to change/simplify the `silence` package's `Query` api, such as adding a `QueryActiveAlerts` method or something, maybe that should be a separate PR?

Reply:
I'm fine with solving this later, but perhaps add a TODO then in the `setAlertStatus` function to somehow dedupe the code with https://github.com/prometheus/alertmanager/blob/master/notify/notify.go#L407-L425 eventually?

@mxinden is currently working on un-munging the many pieces of logic munged together in `main.go`, and this puts this quite significant (and partially duplicated) piece of logic somewhere in `main.go`. I can believe that doing the un-munging in a separate and coordinated effort might even make more sense.